src/contract/nodes.rs

```diff
@@ -671,6 +673,64 @@ impl Transition {
     }
 }
 
+impl MergeRevealed for Genesis {
+    fn merge_revealed_with(&self, other: &Self) -> Result<Self, String> {
+        if self.consensus_commit() == other.consensus_commit() {
+            Ok(Genesis {
+                schema_id: self.schema_id,
+                chain: self.chain.clone(),
+                metadata: self.metadata.clone(),
+                owned_rights: self
+                    .owned_rights
+                    .merge_revealed_with(&other.owned_rights)?,
+                public_rights: self.public_rights.clone(),
+                script: self.script.clone(),
+            })
+        } else {
+            Err(s!("Genesis node missmatch in merged operation"))
+        }
+    }
+}
+
+impl MergeRevealed for Transition {
+    fn merge_revealed_with(&self, other: &Self) -> Result<Self, String> {
+        if self.consensus_commit() == other.consensus_commit() {
+            Ok(Transition {
```

Comment:
Here, and in Genesis + Extension I think we should have (once we will operate on `&mut self` and return `bool`):

```rust
if self..consensus_commit() != other.consensus_commit() {
    return false
}
assert_eq!(self.metadata.merge_revealed(other.metadata), true);
assert_eq!(self.owned_rights.merge_revealed(other.owned_rights), true);
.... // for each other field which is a collection field
```

Reply:
Not asserting inside `Node` implementations. But returning the error message as my below comment.